File: p7.exs
defmodule P7 do  
  def findPrime(limit), do: go(limit, 3, [2], 1)
  
  defp go(limit, _, primes, len) when len >= limit, do: IO.puts(List.last(primes))
  defp go(limit, n, primes, len) do
    n_is_prime? =
      primes
      |> Enum.take(len |> :math.sqrt |> round)
      |> Enum.all?(fn x -> rem(n, x) != 0 end)
    if n_is_prime? do
      go(limit, n + 2, primes ++ [n], len + 1)
    else
      go(limit, n + 2, primes, len)
    end
  end
end

P7.findPrime(10001)

File: p7.py
#my favorite solution
#move from scheme to python because this beauty!

def findPrime(limit):
    import math
    n=3
    primes = [2,]
    while len(primes)<=limit:
        for x in primes[:int(math.sqrt(len(primes)))]:
            if not(n%x): break
        else: primes.append(n)
        n+=2
    return max(primes)

print(findPrime(10001))
